torchtitan/utils.py

```diff
@@ -16,16 +16,17 @@
 import torch.distributed._functional_collectives as funcol
 import torch.distributed.distributed_c10d as c10d
 from torch.distributed.device_mesh import DeviceMesh
+from torchtitan import DEVICE_TYPE, DEVICE_MODULE
 from torchtitan.logging import logger
 
 
 def dist_max(x: Union[int, float], mesh: DeviceMesh) -> float:
-    tensor = torch.tensor(x).cuda()
+    tensor = torch.tensor(x).to(DEVICE_TYPE)
     return funcol.all_reduce(tensor, reduceOp=c10d.ReduceOp.MAX.name, group=mesh).item()
 
 
 def dist_mean(x: Union[int, float], mesh: DeviceMesh) -> float:
-    tensor = torch.tensor(x).cuda()
+    tensor = torch.tensor(x).to(DEVICE_TYPE)
     return funcol.all_reduce(tensor, reduceOp=c10d.ReduceOp.AVG.name, group=mesh).item()
```

Comment:
Is there a way to create tensor on a concrete device with index? Calling `.to(DEVICE_TYPE)` can lead to tensor creation on cuda:0 unintentionally. 
Can be in a separate PR as this seems to be an existing issue.
cc @tianyu-l